moshi/src/main/java/com/squareup/moshi/JsonReader.java

```diff
@@ -348,6 +364,39 @@ public static JsonReader of(BufferedSource source) {
   abstract void promoteNameToValue() throws IOException;
 
   /**
+   * A set of strings to be chosen with {@link #selectName} or {@link #selectString}. This prepares
+   * the encoded values of the strings so they can be read directly from the input source. It cannot
+   * read arbitrary encodings of the strings: if any of a string's characters are unnecessarily
+   * escaped in the source JSON, that string will not be selected. Similarly, if the string is
+   * unquoted or uses single quotes in the source JSON, it will not be selected. Client code that
+   * uses this class should fall back to another mechanism to accommodate this possibility.
+   */
+  static final class Selection {
+    final String[] strings;
+    final List<ByteString> doubleQuoteSuffix;
+
+    public Selection(String[] strings, List<ByteString> doubleQuoteSuffix) {
+      this.strings = strings;
+      this.doubleQuoteSuffix = doubleQuoteSuffix;
+    }
+
+    public static Selection of(String... strings) {
+      try {
+        ByteString[] result = new ByteString[strings.length];
+        Buffer buffer = new Buffer();
+        for (int i = 0; i < strings.length; i++) {
+          BufferedSinkJsonWriter.string(buffer, strings[i]);
+          buffer.readByte(); // Skip the leading double quote (but leave the trailing one).
```

Comment:
Actually maybe that's too big of a behavior change. When you peek you consume the first double quote, right?